cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java

```diff
@@ -141,6 +142,12 @@ public CassandraScheduler(
     public void registered(SchedulerDriver driver,
                            Protos.FrameworkID frameworkId,
                            Protos.MasterInfo masterInfo) {
+        // Return early if registered() has already been called once in the lifetime of this object.
+        if (!isSchedulerRegistered.compareAndSet(false, true)) {
```

Comment:
I also think that calling `reregistered()` should be sufficient in this case.

Also, you are right that reconcile() is called every time we receive an offer. But in a busy cluster, offer cycle can either be delayed OR in worst case the framework might not receive offer for a long time. And, for that duration the framework will live the lie about it's view of world for a longer time v/s the pro-active reconcile() triggered as part of re-registered call that you proposed.

What do you think ?